smart-contract/deploy/gatekeeper.ts

```diff
@@ -1,12 +1,13 @@
 import { addContractToAdmin, sleep, verify, getDeploymentSigner } from "./defender-utils";
-import { ethers } from 'hardhat';
+import { ethers, upgrades } from 'hardhat';
 import { Signer } from '@ethersproject/abstract-signer/src.ts'
 
 async function main() {
     const signer: Signer = await getDeploymentSigner();
+    const signerAddress = await signer.getAddress();
 
     const GatekeeperContractFactory = await ethers.getContractFactory("Gatekeeper", signer!);
-    const gatekeeperContract = await GatekeeperContractFactory.deploy();
+    const gatekeeperContract = await upgrades.deployProxy(GatekeeperContractFactory, [signerAddress], { kind: 'uups', unsafeAllow: ['state-variable-immutable']});
```

Comment:
Instead of the blanket unsafeAllow, should we just use the declaration `/// @custom:oz-upgrades-unsafe-allow state-variable-immutable` in front of the variable? See doc:

https://docs.openzeppelin.com/upgrades-plugins/1.x/faq#why-cant-i-use-immutable-variables

Can you maybe tell me a a sentence when an immutable variable is upgrade-safe?

Reply:
resolved